src/query/storage/m3/cluster_resolver.go

```diff
@@ -43,7 +42,15 @@ const (
 type unaggregatedNamespaceDetails struct {
 	satisfies        unaggregatedNamespaceType
 	clusterNamespace ClusterNamespace
-	retention        time.Duration
+}
+
+type resolvedNamespace struct {
+	ClusterNamespace
+	narrowing narrowing
```

Comment:
I find this name confusing. I don't get the meaning - narrowing what? To my understanding it caries information of time range the namespace covers which sometimes can be before `now`. So maybe we can call it `timeRangeCovered` or smth like that?

Reply:
or maybe just `timeRange` ? It seems to tell storage layer what range to query.

Reply:
It does not tell directly what range to query. It specifies the bounds to which the original query can be narrowed down:
```go
	if !namespace.narrowing.start.IsZero() && namespace.narrowing.start.After(o.StartInclusive) {
		narrowed.StartInclusive = namespace.narrowing.start
	}
	if !namespace.narrowing.end.IsZero() && namespace.narrowing.end.Before(o.EndExclusive) {
		narrowed.EndExclusive = namespace.narrowing.end
	}
```
If you take this semantics into account, then I believe the naming is good here.